pkg/vsphere/session/session.go

```diff
@@ -319,6 +319,28 @@ func (s *Session) Folders(ctx context.Context) *object.DatacenterFolders {
 	return s.folders
 }
 
+// QueryOption looks up the option's values and returns the first one from the
+// ones received on success.
+func (s *Session) QueryOption(ctx context.Context, option string) (string, error) {
+	client := s.Vim25()
+	if client == nil {
+		return "", fmt.Errorf("session client is nil")
+	}
+
+	optMgr := object.NewOptionManager(client, *client.ServiceContent.Setting)
+	opts, err := optMgr.Query(ctx, option)
+	if err != nil {
+		return "", fmt.Errorf("error querying option: %s", err)
+	}
+
+	for _, opt := range opts {
```

Comment:
Maybe instead of the for loop, if len(opt) == 1, return o.Value.  Else, fallthrough to return fmt.Errorf("%d values querying option %q", len(opt), option)